discovery-infra/utils.py

```diff
@@ -23,12 +24,30 @@ def run_command(command, shell=False):
     process = subprocess.run(
         command,
         shell=shell,
-        check=True,
         stdout=subprocess.PIPE,
-        universal_newlines=True,
+        stderr=subprocess.PIPE,
+        universal_newlines=True
     )
-    output = process.stdout.strip()
-    return output
+
+    def _io_buffer_to_str(buf):
+        if hasattr(buf, 'read'):
+            buf = buf.read().decode()
+        return buf
+
+    out = _io_buffer_to_str(process.stdout).strip()
+    err = _io_buffer_to_str(process.stderr).strip()
+
+    if err:
+        raise RuntimeError(
```

Comment:
and if u have something in stderr but command returned 0 ? why to fail? we can't know how others implement there code, u can rely only on exit code

Reply:
if u want custom exception please create one

Reply:
or as i said previously add flag that says something like verify raise_on_failure and set it to check=<this_flag>
and always return output, errorcode i wourld even send stderr to output so if u have error u can just print output